src/Servers/IIS/IIS/src/Core/IISHttpContextOfT.cs

```diff
@@ -21,32 +21,33 @@ public IISHttpContextOfT(MemoryPool<byte> memoryPool, IHttpApplication<TContext>
             _application = application;
         }
 
-        public override async Task<bool> ProcessRequestAsync()
+        public override async Task ProcessRequestAsync()
         {
-            InitializeContext();
-
             var context = default(TContext);
             var success = true;
 
             try
             {
-                context = _application.CreateContext(this);
+                InitializeContext();
```

Comment:
Can this actually throw? If so, do we want to leave success = true? I'm not really sure what PostCompletion() does.

Reply:
InitializeContext would throw if we had a bug in the request initialization (e.g. request url decoding). There is a catch block that sets success=false for this case before calling PostCompletion.